### ISSUE TITLE
error importing ipfs-core from TypeScript "Cannot find namespace 'fs'."

### ISSUE THREAD

**Issue by gobengo** [ipfs/js-ipfs#4080]
- **Version**: 0.14.2

- **Platform**: Darwin MacBook-Pro.local 21.3.0 Darwin Kernel Version 21.3.0: Wed Jan  5 21:37:58 PST 2022; root:xnu-8019.80.24~20/RELEASE_ARM64_T8101 arm64

- **Subsystem**:
- ipfs-core package
  - it's even possible that this is a quirk of how `npm publish` happens, since the error references `node_modules/ipfs-utils/dist/src/files/glob-source.d.ts`

#### Description:

* Here is a reduced test case running on gh actions: https://github.com/gobengo/ipfs-core-esm-test/runs/5951490366?check_suite_focus=true

Note: this issue is similar to https://github.com/ipfs/js-ipfs/issues/4029 , but a different ts error so not the same

#### Steps to reproduce the error:

* start a new typescript project
* `import * as IPFS from 'ipfs-core'`
* try to `tsc build`
* see error

```
bengo@MacBook-Pro ~/dev/ipld-experiments/ipld-web3-storage ⚡  npm run build                                         

> ipld-web3-storage@1.0.0 build
> tsc

node_modules/ipfs-utils/dist/src/files/glob-source.d.ts:10:14 - error TS2503: Cannot find namespace 'fs'.

10     content: fs.ReadStream | undefined;
                ~~


Found 1 error in node_modules/ipfs-utils/dist/src/files/glob-source.d.ts:10
```

I tried clonging js-ipfs repo myself to `npm link` and find a fix, but `npm install` on my m1 mac fails because `npm ERR! response status 404 Not Found on https://node-webrtc.s3.amazonaws.com/wrtc/v0.4.7/Release/darwin-arm64.tar.gz` which is https://github.com/node-webrtc/node-webrtc/issues/680

**Comment by gobengo** [comment]
I was able to [recreate](https://github.com/gobengo/ipfs-core-esm-test/pull/3) this issue only against ipfs-utils, not ipfs-core. So I will open an issue/PR there. 

The good news is I think I also have a patch for ipfs-utils that will fix it (based on npm link testing)

**Comment by callezenwaka** [comment]
Hi, 
Any solution to this yet? Ran into same issue.

**Comment by jamesrwaugh** [comment]
@callezenwaka For now, adding `"sourceMap": true` to tsconfig compilerOptions worked for me.
As per https://github.com/ipfs/js-ipfs/issues/4029#issuecomment-1023028735

**Comment by gobengo** [comment]
reopening to track ipfs-core upgrading to the fixed ipfs-utils@9.0.6 (if needed)
